File: problems/s/stararrangements/stararrangements.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    s = int(input())
    print(f"{s}:")
    for first_line in range(2, s//2 + 2):
        for second_line in [first_line -1, first_line]:
            two_lines = first_line + second_line
            if s % two_lines == 0 or s % two_lines == first_line:
                print(f"{first_line},{second_line}")


if __name__ == '__main__':
    main()

File: problems/s/stararrangements/stararrangements.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int s;

    cin >> s;

    cout << s << ":" << endl;

    for (int first_line=2; first_line<=(s/2+1); first_line++) {
        for (int second_line=first_line-1; second_line<=first_line; second_line++) {
            int two_lines = first_line + second_line; 
            if (s % two_lines == 0 or s % two_lines == first_line) {
                cout << first_line << "," << second_line << endl;
            }
        }
    }
}